controller/node_controller.go

```diff
@@ -1067,7 +1067,11 @@ func (nc *NodeController) deleteOrphans(node *longhorn.Node, diskName string, di
 				if err := nc.ds.DeleteOrphan(orphan.Name); err != nil && !apierrors.IsNotFound(err) {
 					return errors.Wrapf(err, "failed to delete orphan %v", orphan.Name)
 				}
-				diskInfo.OrphanedReplicaDirectoryNames[orphan.Spec.Parameters[longhorn.OrphanDataName]] = longhorn.OrphanCRStatusDeleted
+
+				if _, ok := diskInfo.OrphanedReplicaDirectoryNames[orphan.Spec.Parameters[longhorn.OrphanDataName]]; ok {
+					diskInfo.OrphanedReplicaDirectoryNames[orphan.Spec.Parameters[longhorn.OrphanDataName]] = longhorn.OrphanCRStatusDeleted
```

Comment:
Do you mean we don't need to orphan record deleted records in each diskInfo? then do we still need `longhorn.OrphanCRStatusDeleted`?

Reply:
> Do you mean we don't need to orphan record deleted records in each diskInfo? 

The purpose is to avoid adding extra replica directories in `OrphanedReplicaDirectoryNames` although it does not introduce any impact. Just make the `OrphanedReplicaDirectoryNames` clear and easy to debug.

We still need the `OrphanCRStatusDeleted `. If the node monitor does not finish the collection of the data, we won't hope the node monitor recreates the orphan CRs for the deleted orphaned directories according to the old OrphanedReplicaDirectoryNames list.

Reply:
Actually, we only compare the disk path and some fields with the collected data. Thus, node controller will possibly re-do creation or deletion of orphan CRs. We need a way to avoid the re-creation of the deleted CRs.